src/test/java/com/google/javascript/gents/singleTests/inner_types.ts

```diff
@@ -0,0 +1,28 @@
+
+export class MyClass {
+  /** @export type property*/
+  type: Type1;
+
+  /**
+   * Constructor for MyClass
+   */
+  constructor(data: {type: Type1}) {
+    this.type = data.type;
+  }
+
+  /**
+   * equal function
+   * @export
+   */
+  equals(otherData: MyClass): boolean {
+    return this.type.a === otherData.type.a;
+  }
+}
+
+export type Type1 = {
```

Comment:
I don't know much about how closure handles these. But if they are public types, how will you handle references to them in other code (and can you write a multitest for that?). If they're private, you don't need to export.